src/reanimated2/animations.js

```diff
@@ -354,55 +354,83 @@ export function withDecay(userConfig, callback) {
     'worklet';
     const config = {
       deceleration: 0.998,
+      velocityFactor: 1,
     };
     if (userConfig) {
       Object.keys(userConfig).forEach((key) => (config[key] = userConfig[key]));
     }
 
-    const VELOCITY_EPS = 5;
+    const VELOCITY_EPS = 1;
+    const SLOPE_FACTOR = 0.1;
 
     function decay(animation, now) {
-      const { lastTimestamp, initialVelocity, current, velocity } = animation;
+      const {
+        lastTimestamp,
+        startTimestamp,
+        initialVelocity,
+        current,
+        velocity,
+      } = animation;
 
       const deltaTime = Math.min(now - lastTimestamp, 64);
-      animation.lastTimestamp = now;
-
-      const kv = Math.pow(config.deceleration, deltaTime);
-      const kx = (config.deceleration * (1 - kv)) / (1 - config.deceleration);
-
-      const v0 = velocity / 1000;
-      const v = v0 * kv * 1000;
-      const x = current + v0 * kx;
-
-      animation.current = x;
+      const v =
+        velocity *
+        Math.exp(
+          -(1 - config.deceleration) * (now - startTimestamp) * SLOPE_FACTOR
+        );
+      animation.current =
+        current + (v / (1000 * config.velocityFactor)) * deltaTime;
       animation.velocity = v;
+      animation.lastTimestamp = now;
 
-      let toValueIsReached = null;
-
-      if (Array.isArray(config.clamp)) {
+      if (config.clamp) {
         if (initialVelocity < 0 && animation.current <= config.clamp[0]) {
-          toValueIsReached = config.clamp[0];
+          animation.current = config.clamp[0];
+          return true;
         } else if (
           initialVelocity > 0 &&
           animation.current >= config.clamp[1]
         ) {
-          toValueIsReached = config.clamp[1];
+          animation.current = config.clamp[1];
+          return true;
         }
       }
 
-      if (Math.abs(v) < VELOCITY_EPS || toValueIsReached !== null) {
-        if (toValueIsReached !== null) {
-          animation.current = toValueIsReached;
-        }
-
+      if (Math.abs(v) < VELOCITY_EPS) {
         return true;
       }
     }
 
+    function validConfig() {
+      if (config.clamp) {
+        if (Array.isArray(config.clamp)) {
+          if (config.clamp.length !== 2) {
+            console.error(
+              '[reanimated] clampy array must contain 2 items but is given ' +
```

Comment:
AFAIK we don't use `[reanimated]` tag anywhere else.
Also, I think it should be `clamp array`. You can use string interpolation here as well.

Reply:
👍